src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($day-size + $gap) * $col-count - $gap;
+  gap: $gap;
+  padding: $padding;
+
+  &__day {
+    display: flex;
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid black;
+    background-color: #eee;
+
+    position: relative;
+    text-align: center;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+      transition-duration: 0.5s;
+    }
+
+    @for $number from 1 through $month-length {
```

Comment:
Ensure that the variable $month-length is defined and holds the correct number of days for the month you are styling. Otherwise, the loop will not work as intended.